src/meta/src/hummock/manager/mod.rs

```diff
@@ -901,14 +892,25 @@ where
         };
         let compaction = compaction_guard.deref_mut();
         let start_time = Instant::now();
-        let mut compact_status = VarTransaction::new(
-            compaction
-                .compaction_statuses
-                .get_mut(&compact_task.compaction_group_id)
-                .ok_or(Error::InvalidCompactionGroup(
-                    compact_task.compaction_group_id,
-                ))?,
-        );
+        let compaction_groups: HashSet<_> = self
+            .compaction_group_manager
+            .compaction_groups()
```

Comment:
Clone entire compaction group can be expensive because it may contain many table_ids. This is an existent issue before this PR. Now that we have quite some usage of compaction_groups, let's consider optimize this (maybe in another PR).

Reply:
> existent

I'm sorry. We can call `compaction_group_ids()` here instead. fixed.